src/ray/core_worker/core_worker.cc

```diff
@@ -425,8 +425,13 @@ CoreWorker::CoreWorker(const CoreWorkerOptions &options, const WorkerID &worker_
       memory_store_,
       reference_counter_,
       /*put_in_local_plasma_callback=*/
-      [this](const RayObject &object, const ObjectID &object_id) {
+      [this](const RayObject &object,
+             const ObjectID &object_id,
+             const bool mark_object_failed = false) {
         RAY_CHECK_OK(PutInLocalPlasmaStore(object, object_id, /*pin_object=*/true));
+        if (mark_object_failed) {
```

Comment:
I think we don't need to have `mark_object_failed` boolean: we can always set pending_creation to false whenever `put_in_local_plasma_callback` is called.